src/Chapter4.hs

```diff
@@ -610,7 +624,14 @@ Implement the 'Monad' instance for our lists.
 🕯 HINT: You probably will need to implement a helper function (or
   maybe a few) to flatten lists of lists to a single list.
 -}
+flatten :: List (List a) -> List a
+flatten Empty = Empty
+flatten (Cons (Cons x Empty) xs) = Cons x (flatten xs)
+flatten (Cons _ _)               = Empty
```

Comment:
I'm afraid this implementation is not exactly correct. The way it's implemented, it keeps only singleton lists. Otherwise it returns `Empty` list. To implement `flatter`, try doing the following:

1. Implement list append function first:

```haskell
append :: List a -> List a -> List a
```

2. Implement `flatten` after that, using `append`

```haskell
flatten :: List (List a) -> List a
flatten Empty = ...
flatten (Cons x xs) = ...
```

Reply:
Thank you for suggestion. I changes as followings and used in definition of Applicative and Monad of List.

```
append :: List a -> List a -> List a
append Empty b       = b
append (Cons a as) b = Cons a (append as b) 

flatten :: List (List a) -> List a
flatten Empty = Empty
flatten (Cons x xs) = append x (flatten xs)

```